Code/src/main/java/nl/utwente/viskell/ui/components/ArbitraryBlock.java

```diff
@@ -0,0 +1,87 @@
+package nl.utwente.viskell.ui.components;
+
+import java.util.Optional;
+
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import nl.utwente.viskell.ghcj.GhciSession;
+import nl.utwente.viskell.haskell.type.*;
+import nl.utwente.viskell.ui.ToplevelPane;
+
+public class ArbitraryBlock extends ValueBlock {
+    
+    @FXML private Button rngTrigger; 
+    
+    /**
+     * Constructs a new ArbitraryBlock
+     * @param pane The parent pane this Block resides on.
+     */
+    public ArbitraryBlock(ToplevelPane pane) {
+        super("ArbitraryBlock", pane, pane.getEnvInstance().buildType("Arbitrary a => a"));
+        this.rngTrigger.setOnAction(event -> this.getNextValue(event.hashCode()));
+        this.getNextValue(this.hashCode());
+    }
+
+    private void getNextValue(int seed) {
+        this.setValue("?????");
+        
+        if (! this.output.hasConnection()) {
+            return;
+        }
+
+        // we cannot generate values for complicated types
```

Comment:
If we have a polymorph type like 'Arbitary a => [a]', what values could be generated for that? You don't know enough of 'a' yet.